### PR TITLE
Add int8 support to div operations

### PR DESCRIPTION
**Pull request by pvaneck**
I ran into issues when performing division operations using tf.int8 datatypes, getting errors like:

No OpKernel was registered to support Op 'FloorDiv' with these attrs.
No OpKernel was registered to support Op 'TruncateDiv' with these attrs.

This PR registers int8 as a datatype for these division ops.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Review by rryan — DISMISSED** [review]
The cwise_ops_test.py tests look sufficient to me, but it's not really my area of expertise. @ekelsen did you add me for any particular reason?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @case540, @ekelsen: It has been 20 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @case540, @ekelsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by drpngx** [comment]
```
tensorflow/core/kernels/cwise_op_div.cc:31:60: error: macro "REGISTER9" passed 14 arguments, but takes just 13
           uint16, int8, int16, int64, complex64, complex128);
                                                            ^
tensorflow/core/kernels/cwise_op_div.cc:30:1: error: 'REGISTER9' does not name a type
 REGISTER9(BinaryOp, GPU, "Div", functor::div, float, Eigen::half, double, uint8,
```

**Comment by tensorflowbutler** [comment]
Nagging Assignees @case540, @ekelsen: It has been 22 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @case540, @ekelsen: It has been 37 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rthadur** [comment]
@pvaneck can you please resolve conflicts ?

**Comment by pvaneck** [comment]
Conflicts have been resolved.

**Comment by rthadur** [comment]
> Conflicts have been resolved.

Thank you , @drpngx can you please review

File: tensorflow/core/kernels/cwise_op_div_1.cc [hunk 1]
```diff
@@ -18,20 +18,19 @@ limitations under the License.
...
           bfloat16, complex64, complex128);
 REGISTER5(BinaryOp, CPU, "DivNoNan", functor::div_no_nan, Eigen::half, float,
           double, complex64, complex128);
 
 #if GOOGLE_CUDA
-REGISTER9(BinaryOp, GPU, "Div", functor::div, float, Eigen::half, double, uint8,
-          uint16, int16, int64, complex64, complex128);
-REGISTER4(BinaryOp, GPU, "TruncateDiv", functor::div, uint8, uint16, int16,
-          int64);
+REGISTER5(BinaryOp, GPU, "Div", functor::div, float, Eigen::half, double, uint8, uint16);
```
Comment (josh11b) [inline tensorflow/core/kernels/cwise_op_div_1.cc]: Looks like you are removing registrations for "Div"? These have been moved to cwise_op_div_2? Why was that needed? That seems strange: shouldn't we have all the registrations for "Div" together in one file? If we can't have all these registrations in one file, perhaps we should split more evenly (and have a comment explaining why we needed to split)?

File: tensorflow/core/kernels/cwise_op_div_2.cc [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
```
Comment (josh11b) [inline tensorflow/core/kernels/cwise_op_div_2.cc]: 2019

**Review by josh11b — DISMISSED** [review]

File: tensorflow/core/kernels/cwise_op_div_2.cc (hunk 2 above)
Comment (pvaneck) [reply tensorflow/core/kernels/cwise_op_div_2.cc]: Done

File: tensorflow/core/kernels/cwise_op_div_1.cc (hunk 1 above)
Comment (pvaneck) [reply tensorflow/core/kernels/cwise_op_div_1.cc]: Hey Josh. Thanks for the review.

I would have kept it in one file, however, the macro for registering this many types no longer exists (REGISTER10), with the removal and basis for using multiple files stemming from this commit: https://github.com/tensorflow/tensorflow/commit/a373b0f6bf3dd3ebd80ad9c72485de9c37c815ae

I have been using https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/cwise_op_mul_1.cc and
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/cwise_op_mul_2.cc as a guideline.

I only split the operation that tried to register 10+ types which was only the GPU 'Div' registration.

Comment (josh11b) [reply tensorflow/core/kernels/cwise_op_div_1.cc]: I much prefer the div operations be in one file. Please split so that e.g. Div & TruncateDiv are in one file and RealDiv and DivNoNan is in the other (or whatever else makes sense given what code is shared between operations).

**Comment by petewarden** [comment]
Sorry for the slow response, but I'm not familiar enough with this area to review it. @drpngx, are you able to take over the review?

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by rthadur** [comment]
@pvaneck can you please resolve conflicts and address Josh's review comments.

**Comment by rthadur** [comment]
closing this PR as contrib folder will be depricated in 2.0, thank you.
CC @mihaimaruseac

**Comment by rryan** [comment]
@rthadur note this PR affects core, it just has a single file touching contrib. It could simply be reverted. 

However, the PR is stalled. @pvaneck are you still interested in adding int8 support or should we close this? So sorry it's taken this long :(.

**Comment by mihaimaruseac** [comment]
Removing the contrib file while solving conflicts or reopening a new PR are both good alternatives.

**Comment by rryan** [comment]
I guess we can close until we hear from @pvaneck. Feel free to re-open!

Thanks again for your work, and sorry it's taken so incredibly long.
